lib/metasploit/framework/sap_solman/client.rb

```diff
@@ -0,0 +1,224 @@
+          soap_body << "\t\t<agentName>#{agent_name}</agentName>\r\n"
+          soap_body << "\t\t<scriptName>#{script_name}</scriptName>\r\n"
+          soap_body << "\t</adm:#{script_action}>\r\n"
+          send_soap_request(soap_body, solman_path)
+        end
+
+        # Stop script in agent
+        def stop_script_in_agent(agent_name, script_name, solman_path)
+          script_action(agent_name, script_name, 'stopScript', solman_path)
+        end
+
+        # Delete script in agent
+        def delete_script_in_agent(agent_name, script_name, solman_path)
+          script_action(agent_name, script_name, 'deleteScript', solman_path)
+        end
+
+        # Get connected agents info
+        def make_agents_array(solman_path)
+          agents = []
+          all_agent_info = send_soap_request('<adm:getAllAgentInfo />', solman_path)
+          response_xml = Nokogiri::XML(all_agent_info.body)
+          response_xml.css('return').each do |agent|
+            os_name = ''
+            java_version = ''
+            agent.css('systemProperties').each do |system_properties|
+              case system_properties.at_xpath('key').content
+              when 'os.name'
+                os_name = system_properties.at_xpath('value').content
+              when 'java.version'
+                java_version = system_properties.at_xpath('value').content
+              end
+            end
+            agents.push({
+              serverName: agent.at_xpath('serverName').content,
+              hostName: agent.at_xpath('hostName').content,
+              instanceName: agent.at_xpath('instanceName').content,
+              osName: os_name,
+              javaVersion: java_version
+            })
+          end
+          agents
+        end
+
+        # For print agents array
+        def make_pretty_table(array)
+          pretty_string = "+-#{@columns.map { |_, g| '-' * g[:width] }.join('-+-')}-+"
+          pretty_string += "\n| #{@columns.map { |_, g| g[:label].ljust(g[:width]) }.join(' | ')} |"
+          pretty_string += "\n+-#{@columns.map { |_, g| '-' * g[:width] }.join('-+-')}-+"
+          array.each do |line|
+            str = line.keys.map { |k| line[k].ljust(@columns[k][:width]) }.join(' | ')
+            pretty_string += "\n| #{str} |"
+          end
+          pretty_string += "\n+-#{@columns.map { |_, g| '-' * g[:width] }.join('-+-')}-+"
+          pretty_string
+        end
+
+        # Pretty print connected agents array
+        def pretty_agents_table(agents)
+          @agent_labels = {
+            serverName: 'Server Name',
+            hostName: 'Host Name',
+            instanceName: 'Instance Name',
+            osName: 'OS Name',
+            javaVersion: 'Java Version'
+          }
+          @columns = @agent_labels.each_with_object({}) do |(col, label), h|
+            h[col] = { label: label, width: [agents.map { |g| g[col].size }.max, label.size].max }
+          end
+          make_pretty_table(agents)
+        end
```

Comment:
We actually have an existing table-printing API in [`Rex::Text::Table`](https://github.com/rapid7/rex-text/blob/master/lib/rex/text/table.rb) that should be used here. Here's a pretty simple example showing how it's used, there's quite a few more if you just search for `Rex::Text::Table.new`: https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/enum_ad_groups.rb#L72.

Reply:
Update method pretty_agents_table in library
https://github.com/rapid7/metasploit-framework/pull/14924/files#diff-ce39e51fed07414ab05f2d911e01956e88010187f1aec059af4a2183dec092f8R179-R192